src/commonMain/kotlin/baaahs/sm/brain/BrainManager.kt

```diff
@@ -106,10 +106,14 @@ class BrainManager(
         }
 
         val existingController = activeBrains[brainId]
-        if (existingController != null) {
+        if (existingController != null && existingController.lastError == null) {
```

Comment:
Maybe this should check the address too?
```kt
&& existingController.brainAddress != brainAddress
```

Reply:
yes this is probably the best way to determine the brain should be re-initialized instead of relying on an error